### ISSUE TITLE
Angular template versioning

### ISSUE THREAD

**Issue by colin-higgins** [Particular/ServicePulse#212]
task-force @colin-higgins @petersgiles
### Goal:

Introduce template versioning with a query string to ensure users have a smooth upgrade experience whenever we release new versions of ServicePulse.
### Plan:

Shared function that uses an angular.constant to tack the version number onto the url for the template.
We could also have a debug flag that refreshes the template every time in development.

**Comment by petersgiles** [comment]
Yes to this! I'm in

**Comment by petersgiles** [comment]
@colin-higgins did you have a specific way of doing this (using an existing framework)?

**Comment by colin-higgins** [comment]
```
angular.module('sp').value('applyVersion', function (templateUrl) { return templateUrl + '?v=1.3.5'; });
```

That is oversimplified obviously, but it's approximately what I have done before.

**Comment by johnsimons** [comment]
Shouldn't we handle this using the normal expiration headers in http?

@SimonCropp has solved this nicely in docs, majority of time it returns 304s.

**Comment by colin-higgins** [comment]
@johnsimons The reason I would argue for the query string method is that it's tied to the version and automatically handles any upgrades seamlessly. Otherwise it leaves the cache alone. How would you implement the expiration headers?

**Comment by johnsimons** [comment]
Sounds good @colin-higgins, your way will work best especially if users host it in IIS where we don't control the headers.

**Comment by WilliamBZA** [comment]
Blast from the past, but a first step to this will finally be released in 1.25.0.
